src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -140,7 +142,7 @@ static string GenerateTypesQuery(bool withRange, bool withEnum, bool withEnumSor
   ELSE 0
 END AS elemoid,
 CASE
-  WHEN a.typtype='d' AND a.typcategory='A' THEN 4                  /* Domains over arrays last */
+  WHEN a.typtype='d' {(decimal.Parse(version.Major + "." + version.Minor) < 8.4m ? "" : "AND a.typcategory='A'")}  THEN 4                  /* Domains over arrays last */
```

Comment:
See my comments above. But for reference, `Version` actually has really nice support for comparisons:
```c#
var a = new Version(10, 4);
var b = new Version(8, 4);

if (a < b) 
    Console.WriteLine("Yes! a < b");
else
    Console.WriteLine("Nope! a >= b");

// prints: Nope! a >= b

```